lib/matplotlib/colorbar.py

```diff
@@ -578,13 +580,11 @@ def _reset_locator_formatter_scale(self):
         """
         self.locator = None
         self.formatter = None
-        if (isinstance(self.norm, colors.LogNorm)
-                and self._use_auto_colorbar_locator()):
```

Comment:
Why can that be left out?

Reply:
I think we want the axes to be logarithmic in this case, but I agree its open to interpretation:

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.colors as mcolors

z = np.arange(24).reshape(4, 6) + 1
z = np.power(2, z)
fig, axs = plt.subplots(1, 2, constrained_layout=True)
ax = axs[0]
cs = ax.contourf(z, levels=np.power(10., [1, 2, 6.5, 10] ), norm=mcolors.LogNorm())
fig.colorbar(cs, ax=ax, spacing='proportional')
ax = axs[1]
cs = ax.contourf(z, levels=np.power(10, [1, 2, 6.5, 10]), norm=mcolors.LogNorm())
fig.colorbar(cs, ax=ax, spacing='uniform')

plt.show()
```

## With this change:

![Figure_1](https://user-images.githubusercontent.com/1562854/57185939-b142d480-6e8a-11e9-8e32-04eb37a6288d.png)

## Without this change:

![Figure_1Wrong](https://user-images.githubusercontent.com/1562854/57185960-31693a00-6e8b-11e9-861a-97beae20e0f1.png)

## Before this PR:

![Figure_1Old](https://user-images.githubusercontent.com/1562854/57185946-e3eccd00-6e8a-11e9-9c0e-efb2ae9c59be.png)

Reply:
👍 Looks reasonable.